cmd/influx_tsm/converter.go

```diff
@@ -68,32 +70,32 @@ func (c *Converter) Process(iter KeyIterator) error {
 
 		// If we have a max file size configured and we're over it, start a new TSM file.
 		if w.Size() > c.maxTSMFileSize {
-			if err := w.WriteIndex(); err != nil && err != tsm1.ErrNoValues {
-				return err
-			}
-			if err := w.Close(); err != nil {
-				return err
-			}
-
-			w, err = c.nextTSMWriter()
-			if err != nil {
-				return err
-			}
-			atomic.AddUint64(&TsmFilesCreated, 1)
+			w = nil
```

Comment:
Should really write the index and close here or otherwise the process will accumulate a lot of heap space because the `directIndex` is still reference and not written.